bin/pyenv-update

```diff
@@ -55,7 +55,13 @@ update_repo() {
   info "Updating $1..."
   verify_repo "$1" &&
   # pyenv-installer makes the repos shallow, so tags are not fetched by default
-  ( cd "${repo}" && git pull --tags --no-rebase --ff "${REMOTE}" "${BRANCH_CHOICE}" )
+  # Use git fetch & merge instead of git pull --ff-only to play nice with old
+  # git.
+  (
+    cd "${repo}" && \
+    git fetch --tags "${REMOTE}" "${BRANCH_CHOICE}" && \
+    git merge --ff-only "${REMOTE}" "${BRANCH_CHOICE}"
```

Comment:
`--ff-only` is not equivalent to `--ff` and will cause a failure if there are local changes -- exactly what we try to avoid.
I'll change that. Tested it to work in a Centos 7 Docker image.

Reply:
Ah, I was only testing with uncommitted local changes. Thanks!